framework/PostProcessors/Metric.py

```diff
@@ -0,0 +1,293 @@
+      if hasattr(currentInput, 'type'):
+        inputType = currentInput.type
+
+      if isinstance(currentInput, Files.File):
+        self.raiseAWarning(self, "Input type '", inputType, "' is not yet implemented. This input will be skipped")
+      elif inputType == 'HDF5':
+        self.raiseAWarning(self, "Input type '", inputType, "' is not yet implemented. This input will be skipped")
+      elif inputType == 'PointSet':
+        for feature in self.features:
+          if feature in currentInput.getParaKeys('input'):
+            if feature in inputDict['features'].keys():
+              self.raiseAnError(IOError, "Same feature variable '", feature, "' is found in multiple input objects")
+            inputDict['features'][feature] = currentInput.getParam('input', feature, nodeId = 'ending')
+          elif feature in currentInput.getParaKeys('output'):
+            if feature in inputDict['features'].keys():
+              self.raiseAnError(IOError, "Same feature variable '", feature, "' is found in multiple input objects")
+            inputDict['features'][feature] = currentInput.getParam('output', feature, nodeId = 'ending')
+        if self.targets:
+          for target in self.targets:
+            if target in currentInput.getParaKeys('input'):
+              if target in inputDict['targets'].keys():
+                self.raiseAnError(IOError, "Same target variable '", target, "' is found in multiple input objects")
+              inputDict['targets'][target] = currentInput.getParam('input', target, nodeId = 'ending')
+            elif target in currentInput.getParaKeys('output'):
+              if target in inputDict['targets'].keys():
+                self.raiseAnError(IOError, "Same target variable '", target, "' is found in multiple input objects")
+              inputDict['targets'][target] = currentInput.getParam('output', target, nodeId = 'ending')
+      elif inputType == 'HistorySet':
+        self.dynamic = True
+        self.raiseAnError(IOError, "Metric can not process HistorySet, because this capability is not implemented yet")
+
+    unidentifiedVars = set(self.features) - set(inputDict['features'].keys())
+    if len(unidentifiedVars) > 0:
+      self.raiseAnError(IOError, "Features: '", str(unidentifiedVars), "' could not be found in the input objects")
+    unidentifiedVars = set(self.targets) - set(inputDict['targets'].keys())
+    if len(unidentifiedVars) > 0:
+      self.raiseAnError(IOError, "Targets: '", str(unidentifiedVars), "' could not be found in the input objects")
+
+    return inputDict
+
+  def initialize(self, runInfo, inputs, initDict) :
+    """
+      Method to initialize the pp.
+      @ In, runInfo, dict, dictionary of run info (e.g. working dir, etc)
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, dictionary with initialization options
+    """
+    PostProcessor.initialize(self, runInfo, inputs, initDict)
+    self.__workingDir = runInfo['WorkingDir']
```

Comment:
do we really need this? I think the base class does this already right?

Reply:
I think we need this. The base class does not set it up. I can move this to PostProcessor.py under PostProcessors folder, or I can move it to the BaseClasses.py. What do you think?

Reply:
I would move this into the PP base class.

Thanks

Reply:
moved to the PP base class

Reply:
The name self.__workingDir is renamed to self._workingDir. This is because self.__workingDir can not be recognized by the subclasses. To be honest, I do not know why.

Reply:
because __ indicates "pseudo"-private attributes (in reality there is a way to access them but it is quite ugly).